raiko-guest/src/one_shot.rs

```diff
@@ -0,0 +1,227 @@
+pub const PRIV_KEY_FILENAME: &str = "priv.key";
+
+pub async fn one_shot(global_opts: GlobalOpts, args: OneShotArgs) -> Result<()> {
+    if args.no_sgx {
+        eprintln!("Running without SGX");
+    }
+
+    if !is_bootstrapped(&global_opts.secrets_dir) {
+        bail!("Application was not bootstrapped. Bootstrap it first.")
+    }
+
+    let path_str = args.blocks_data_file.to_string_lossy().to_string();
+    let block_no = u64::from_str(&String::from(
+        args.blocks_data_file
+            .file_prefix()
+            .unwrap()
+            .to_str()
+            .unwrap(),
+    ))
+    .unwrap();
+    eprintln!("Reading input file {} (block no: {})", path_str, block_no);
+
+    // TODO FIXME
+    let block_header_hash_str = String::from("foobar");
+    // let block_header_hash_str = get_data_to_sign(path_str, block_no).await?;
+    eprintln!("Data to be signed: {}", block_header_hash_str);
+
+    let public_key = sgx_sign(global_opts.secrets_dir, block_header_hash_str)?;
+
+    if !args.no_sgx {
+        print_sgx_info(public_key)?
+    }
+
+    Ok(())
+}
+
+pub fn bootstrap(global_opts: GlobalOpts) -> Result<()> {
+    // TODO REMOVEME temporary debug
+    for path in fs::read_dir("/secrets").unwrap() {
+        println!("Entry: {}", path.unwrap().path().display())
+    }
+
+    let (next_private_key, next_public_key, _secp) = generate_new_keypair(global_opts.secrets_dir)?;
+
+    println!("Next private key: 0x{}", next_private_key.display_secret());
+    println!("Next public key: 0x{}", next_public_key);
+    // TODO REMOVEME temporary debug
+    let paths = fs::read_dir("/secrets").unwrap();
+    for path in paths {
+        println!("Entry: {}", path.unwrap().path().display())
+    }
+    Ok(())
+}
+
+fn generate_new_keypair(
+    secrets_dir: PathBuf,
+) -> Result<(SecretKey, PublicKey, Secp256k1<All>), Error> {
+    let priv_key_path = secrets_dir.join(PRIV_KEY_FILENAME);
+    let secp = Secp256k1::new();
+    let (priv_key, public_key) = secp.generate_keypair(&mut OsRng);
+    fs::write(priv_key_path, SecretKey::secret_bytes(&priv_key))?;
+    Ok((priv_key, public_key, secp))
+}
+
+fn is_bootstrapped(secrets_dir: &PathBuf) -> bool {
+    let priv_key_path = secrets_dir.join(PRIV_KEY_FILENAME);
+    priv_key_path.is_file() && priv_key_path.metadata().unwrap().permissions().readonly() == false
+}
+
+// TODO verify what should be format of the data to be signed
+async fn get_data_to_sign(path_str: String, block_no: u64) -> Result<String> {
+    let init: Init<zeth_lib::EthereumTxEssence> = parse_to_init(path_str, block_no).await?;
+    let input = init.clone().into();
+    let output = TaikoBlockBuilder::build_from(&TAIKO_MAINNET_CHAIN_SPEC, input)
+        .expect("Failed to build the resulting block");
+    let block_header_hash = output.hash();
+    let block_header_hash_str = block_header_hash.to_string();
+    Ok(block_header_hash_str)
+}
```

Comment:
Please help me on this: we need to clarify what is the format of the data that needs to actually be signed.